`mkdir`: make `mkdir` public and document it

KAAtheWiseGit opened the pull request:
- Made the `mkdir` function public.  Added documentation, describing the behavior and parameters
- Moved the GNU dot-stripping behavior from `exec` to `mkdir`.

KAAtheWiseGit commented:
@tertsdiepraam, opened the PR.  It doesn't have much as of now, since I haven't made `mkdir` public.

sylvestre commented on src/uu/mkdir/src/mkdir.rs:
```diff
@@ -136,10 +135,22 @@ pub fn uu_app() -> Command {
+///
+/// ## Options
+///
+/// * `recurisve` --- create parent directories for the given dirs, if they do
```
```suggestion
/// * `recursive` --- create parent directories for the given dirs, if they do
```

KAAtheWiseGit commented:
Rewrote the commit: 
- Made `mkdir` public instead of `exec`.
- Completed the documentation GNU behavior note.
- Moved the GNU patch into `mkdir`.

Changed the PR name/description appropriately.

KAAtheWiseGit commented:
I think the PR is ready for review.

tertsdiepraam commented on src/uu/mkdir/src/mkdir.rs:
```diff
@@ -141,21 +141,34 @@ pub fn uu_app() -> Command {
+/// ## Trailing dot
+///
+/// To match the GNU behavior, a path with the last directory being a single dot
+/// (like `some/path/to/.`) is created (with the dot stripped).
```
Great comment here!

tertsdiepraam reviewed (approved):
I won't merge this myself because I'm reviewing from the GH app, but this looks great!

cakebaker commented:
Thanks :)
